cmd/crc/cmd/root.go

```diff
@@ -96,15 +98,32 @@ func setProxyDefaults() {
 	httpProxy := config.GetString(cmdConfig.HTTPProxy.Name)
 	httpsProxy := config.GetString(cmdConfig.HTTPSProxy.Name)
 	noProxy := config.GetString(cmdConfig.NoProxy.Name)
+	proxyCAFile := config.GetString(cmdConfig.ProxyCAFile.Name)
 
-	proxyConfig, err := network.NewProxyDefaults(httpProxy, httpsProxy, noProxy)
+	proxyCAData, err := getProxyCAData(proxyCAFile)
+	if err != nil {
+		exit.WithMessage(1, fmt.Sprintf("not able to read proxyCAFile %s: %v", proxyCAFile, err.Error()))
+	}
+
+	proxyConfig, err := network.NewProxyDefaults(httpProxy, httpsProxy, noProxy, proxyCAData)
 	if err != nil {
 		exit.WithMessage(1, err.Error())
 	}
 
 	if proxyConfig.IsEnabled() {
-		logging.Debugf("HTTP-PROXY: %s, HTTPS-PROXY: %s, NO-PROXY: %s", proxyConfig.HTTPProxyForDisplay(),
-			proxyConfig.HTTPSProxyForDisplay(), proxyConfig.GetNoProxyString())
+		logging.Debugf("HTTP-PROXY: %s, HTTPS-PROXY: %s, NO-PROXY: %s, proxyCAFile: %s", proxyConfig.HTTPProxyForDisplay(),
+			proxyConfig.HTTPSProxyForDisplay(), proxyConfig.GetNoProxyString(), proxyCAFile)
 		proxyConfig.ApplyToEnvironment()
 	}
 }
+
+func getProxyCAData(proxyCAFile string) (string, error) {
+	if proxyCAFile == "" {
+		return "", nil
+	}
+	proxyCACert, err := ioutil.ReadFile(proxyCAFile)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimRight(string(proxyCACert), "\r\n"), nil
```

Comment:
Not clear what the `strings.TrimRight` call is for? Is this Windows specific?

Reply:
In windows if user open the cert file in notepad or any windows application it does add the `\r\n` trailing so this will just to add extra layer on our side to get the cert data in correct way.

Reply:
I'd move that to a helper at least, and probably add a comment that it's mostly needed on Windows. Why is this removing them completely rather than changing them to `\n`? If the removal is desirable, why not remove `\n` on linux as well?

Reply:
@cfergeau It does remove the endline for linux and windows both, I will put a comment, shouldn't need a helper function just to perform string operation using standard strings package. https://play.golang.org/p/0gKkIhLLxij

Reply:
> It does remove the endline for linux and windows both

Ah, right, "\r\n" is a list of characters to remove, not a substring to remove.

> shouldn't need a helper function just to perform string operation using standard strings package.

This makes the code much easier to read.
Compare this line of thought:

`strings.TrimRight(string(proxyCACert), "\r\n")` What is `TrimRight` doing again? Ah yes, removing characters from a string. Ok these characters are "\r\n" which are end of line characters, so it's removing the end of lines from the string (and this assumes the reader knows that \r is used for end of line on Windows, and \n on all platforms)

with:

`removeLineEndings(string(proxyCACert))` ok, they remove the end of line markers from the string

Reply:
Wondering if removing these line endings is the right thing to do. https://serverfault.com/questions/466683/can-an-ssl-certificate-be-on-a-single-line-in-a-file-no-line-breaks says not to remove line endings from a PEM file (which is what openshift wants)

Reply:
@cfergeau this is not removing the intermediate end line `\n` but only remove if the file end have more new lines something like.
```
-----BEGIN CERTIFICATE-----
Zc8uiGg99pmd008E2cmGErqNzPclK9INbPjwxl9zIgTcXpF/UtBVVS1Z7Xo83qfs
GMPdMzYt3F+hQpQYLhlGF+5Jf2x6Zb1zjT/aM3GMdGi+6OPV+YHl/wgUe45N6kRu
DZnVL167+W3l2nD+mShO/7xqwHiZuz0GHyBHRp5i43blH0vg67sJ8guN81papupY
2v78FcvR8j0ELfgyehtWpjF3vzKSq/rY2sMXTYzSPqMeksseHNhSMYU6Ww9h9pyM
aVnw9vahqf7nKHHcC2VRTUgkQfn9yDmmBOo0nQ8Xgfpd65/PaxVfBnuKfEkXBfpM
-----END CERTIFICATE-----
<multiple new lines here>
```

Reply:
Ah right, sorry again, I really should read carefully what `TrimRight` is doing :( I would still use a helper for this (trimTrailingEOL) to make the code self documenting..
One more question: it seems we assume we may get `\r\n` at the end of the file, but that all other EOL must be `\n`? I'd personally expect that we don't deal with `\r` at all, or that we deal with it everywhere (replace `\r\n` with `\n`, and then remove trailing `\n`

Reply:
@cfergeau the assumption we are having here is at the file end only user might add new lines by mistake and not in between so default endline now doesn't change to `\r\n` only the newly added on if any for windows.

Reply:
I'd go for "all or nothing", either we remove all `\r` + trailing new lines, or we don't remove any `\r`. Handling them in some specific cases but not others is unexpected